### PR TITLE
Only run build docker image on merge

### PR DESCRIPTION
**Pull request by paurakhsharma**
## Description
Only run build-docker-image on pull request merge

## Related Issue
#844

## Motivation and Context
Make drone faster

## How Has This Been Tested?
🤖

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
Looks like there is a yarn install missing ...

```
+ cd /var/www/owncloud/phoenix
+ yarn run lint
yarn run v1.15.2
$ eslint src apps/*/src tests build --ext vue --ext js --color
/bin/sh: 1: eslint: not found
error Command failed with exit code 127.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Comment by paurakhsharma** [comment]
> Looks like there is a yarn install missing ...

Yes, I just moved wait-for-phoenix before lint-test so, that yarn gets installed before lint-test is run.

File: .drone.yml [hunk 1]
```diff
@@ -12,6 +12,20 @@ pipeline:
...
     commands:
       - docker build -t owncloud/phoenix:${DRONE_COMMIT_SHA}-${DRONE_BUILD_NUMBER} .
+    when:
+      event: [ push ]
+
+  install-phoenix:
+    image: owncloudci/php:${PHP_VERSION=7.1}
+    pull: true
+    commands:
+      - git clone https://github.com/owncloud/owncloud-design-system.git
```
Comment (DeepDiver1975) [inline .drone.yml]: we  don't need this line anymore

File: .drone.yml [hunk 2]
```diff
@@ -12,6 +12,20 @@ pipeline:
...
+      - cd owncloud-design-system
```
Comment (DeepDiver1975) [inline .drone.yml]: we  don't need this line anymore

File: .drone.yml [hunk 3]
```diff
@@ -12,6 +12,20 @@ pipeline:
...
+      - yarn install && yarn build:system
```
Comment (DeepDiver1975) [inline .drone.yml]: we  don't need this line anymore

File: .drone.yml [hunk 4]
```diff
@@ -12,6 +12,20 @@ pipeline:
...
+      - cd ..
```
Comment (DeepDiver1975) [inline .drone.yml]: we  don't need this line anymore

File: .drone.yml (hunk 2 above)
Comment (paurakhsharma) [reply .drone.yml]: Oh yes, We use npm package now.

**Comment by paurakhsharma** [comment]
> Please remove the commented relics from ancient times .... THX

Oh yes, We use npm package now.

**Comment by individual-it** [comment]
drone fails with `fatal: unable to access 'https://github.com/owncloud/phoenix.git/': The requested URL returned error: 503`

**Pull request merged**